fpga_interchange/arch.h

```diff
@@ -1303,29 +573,33 @@ struct Arch : ArchAPI<ArchRanges>
         return range;
     }
 
-    Loc getPipLocation(PipId pip) const override
+    Loc getPipLocation(PipId pip) const final
     {
         Loc loc;
         get_tile_loc(pip.tile, &loc);
         loc.z = 0;
         return loc;
     }
 
-    uint32_t getPipChecksum(PipId pip) const override { return pip.index; }
+    uint32_t getPipChecksum(PipId pip) const final { return pip.index; }
 
-    WireId getPipSrcWire(PipId pip) const override
+    WireId getPipSrcWire(PipId pip) const final NPNR_ATTRIBUTE(__always_inline__)
```

Comment:
can you create a `NPNR_ALWAYS_INLINE` macro? that way someone can do a MSVC version if they care enough

Reply:
Will do.

Reply:
Fixed